src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Extensions.Internal
+{
+    internal class RuntimeEnvironment
+    {
+        public RuntimeEnvironment()
+        {
+            OperatingSystemVersion = Native.PlatformApis.GetOSVersion();
+
+            RuntimeType = GetRuntimeType();
+        }
+
+        public string OperatingSystemVersion { get; }
```

Comment:
We should be able to remove the `GetRID` method entirely. I'll clean up the one use of it from Scaffolding. If we remove the use from `StartupExceptionPage` we'd only have native code usage in non-shipping code.